### PR TITLE
Suppression of the logger messages.

### PR DESCRIPTION
**Pull request by drivanov**
Implement the ability to suppress/unsuppress messages sent to the logger. 
This feature can be very useful for suppressing error messages from tests that should fail.

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc [hunk 1]
```diff
@@ -44,14 +47,37 @@ void Logger::log(Severity severity, const char* msg) noexcept {
...
+  }
+}
+
+void Logger::unsuppressLoggerMsgs(Severity severity) {
+  if (checkSeverity(severity, "unsuppressLoggerMsgs")) {
+    suppressedMsg_ &= (-1) ^ (1 << (int)severity);
+  }
+}
+
+bool Logger::checkSeverity(Severity severity, const char* msg) noexcept {
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:65]: Let's rename this to: isValidSeverity

**Review by bixia1 — COMMENTED** [review]
Please also squash the commits, after the depend-on PR is merged.

File: tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc (hunk 1 above)
Comment (drivanov) [reply tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc]: Done

**Review by bixia1 — APPROVED** [review]

File: tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc [hunk 2]
```diff
@@ -44,14 +47,42 @@ void Logger::log(Severity severity, const char* msg) noexcept {
...
-      LOG(FATAL) << name_ << "Got unknown severity level " << int(severity)
-                 << " from TensorRT: " << msg;
-      break;
+      std::stringstream ss;
+      ss << name_ << "Got unknown severity level " << int(severity);
+      if (msg) {
+        LOG(FATAL) << ss.str() << " from TensorRT: " << msg;
+      } else {
+        LOG(ERROR) << ss.str();
+      }
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:82]: Tensorflow has a new check that prevents new code with LOG(FATAL).
Can we just remove "default" to make this an exhaustive switch statement?

**Review by bixia1 — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc (hunk 2 above)
Comment (drivanov) [reply tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:82]: DONE

**Review by bixia1 — APPROVED** [review]

**Pull request merged**